src/lock.js

```diff
@@ -68,4 +69,42 @@ class Lock {
   }
 }
 
-export const lock = new Lock();
+class GlobalLock {
+  constructor() {
+    this.wallets = {};
+    this.globalLock = new Lock();
+  }
+
+  start(walletId) {
+    if (!this.wallets[walletId]) {
+      this.wallets[walletId] = new Lock();
+    }
+    return this.wallets[walletId];
+  }
+
+  delete(walletId) {
+    if (this.wallets[walletId]) {
+      delete this.wallets[walletId];
+    }
+  }
+
+  get(walletId) {
+    if (!this.wallets[walletId]) {
+      return this.start(walletId);
```

Comment:
suggestion (non-blocking): remove `start` method. We can have the line `this.wallets[walletId] = new Lock();` here and remove the `start` method. What do you think?

Reply:
Done